### PR TITLE
add two helper methods to build ObjectId related eq filters

### PR DESCRIPTION
**Pull request by kroepke**
## Description

we already have helpers for the _id field, but sometimes we store objectid 
references in other fields and it's unwieldy to create objectid instances all the time

/nocl helper methods

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Review by kingzacko1 — APPROVED** [review]

**Review by danotorrey — APPROVED** [review]

File: graylog2-server/src/test/java/org/graylog2/database/utils/MongoUtilsTest.java [hunk 1]
```diff
@@ -46,14 +46,19 @@ class MongoUtilsTest {
 
     private record DTO(@Id @org.mongojack.ObjectId String id, String name) implements MongoEntity {}
 
+    private record DTORef(@Id @org.mongojack.ObjectId String id, @org.mongojack.ObjectId ObjectId refId,
```
Comment (thll) [inline graylog2-server/src/test/java/org/graylog2/database/utils/MongoUtilsTest.java:49]: I think the `@ObjectId` annotation only makes sense for fields of type `String`?

Comment (kroepke) [reply graylog2-server/src/test/java/org/graylog2/database/utils/MongoUtilsTest.java:49]: Yes, copy'n'waste mistake, I'll fix

Comment (kroepke) [reply graylog2-server/src/test/java/org/graylog2/database/utils/MongoUtilsTest.java:49]: Actually, I think it's correct (note it's the same on the other record):
Per mongojack ObjectId javadoc
```
Indicates that the property should be converted to an ObjectId before being stored in the database, and back to whatever type it is afterwards. Use this if you want an ID to have a String or byte[] value, but you want it stored in the database as an ObjectId.
```

Comment (kroepke) [reply graylog2-server/src/test/java/org/graylog2/database/utils/MongoUtilsTest.java]: Nevermind, confused the parameters :)

**Pull request merged**